test/spootnik/reporter/impl_test.clj

```diff
@@ -23,6 +31,90 @@
       (is (.contains ^String metrics "TYPE timer_promtest summary")))
     (component/stop reporter)))
 
+(def sample-port 9058)
+
+(deftest prometheus-server-test
+  (let [port     sample-port
+        system   (map->Reporter {:prometheus {:port port}
+                                 :metrics    {:reporters {:prometheus {}}}})
+        reporter (component/start system)
+        jvm-help "# HELP jvm_threads_current Current thread count of a JVM"]
+    (Thread/sleep 5000)
+    (testing (str "metrics served on port " port)
+      (let [res  @(http/get (format "http://localhost:%s/metrics" port))
+            body (:body (update res :body bs/to-string))]
```

Comment:
You could `(some-> res :body bs/to-string (str/includes? jvm-help))` since you throw away the response

Reply:
Good point @mpenet.